docroot/modules/custom/sitenow_people/sitenow_people.module

```diff
@@ -125,15 +130,117 @@ function sitenow_people_form_alter(&$form, FormStateInterface $form_state, $form
+      ) {
+        $form['#prefix'] = '<div id="form-ajax-wrapper">';
+        $form['#suffix'] = '</div>';
+        // Add custom process callback to alter exposed filters.
+        $form['settings']['override']['exposed_filters']['#process'][] = '_sitenow_people_process_people_block_filters';
+      }
+      break;
+  }
+}
+
+/**
+ * Custom process block form elements.
+ *
+ * @param array $element
+ *   The current block element.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ *
+ * @return array
+ *   The processed block element.
+ */
+function _sitenow_people_process_people_block_filters(array $element, FormStateInterface $form_state, array &$form) {
+  /** @var \Drupal\layout_builder\SectionComponent $component */
+  $component = $form_state
+    ->getFormObject()
+    ->getCurrentComponent();
+  $block = $component
+    ->getPlugin();
+
+  switch ($block->getPluginId()) {
+    case 'views_block:people_list_block-list_card':
+
+      // Keep exposed filters open while filters have values.
+      foreach (Element::children($element) as $field_name) {
+        $default = $element[$field_name]['#default_value'];
+        $filter = $form_state->getValue([
+          'settings',
+          'override',
+          'exposed_filters',
+          $field_name,
+        ]);
+        if ($filter || !empty($default)) {
+          $element['#open'] = TRUE;
+          break;
+        }
+      }
+
+      if (isset($element['field_person_types_target_id'])) {
+        _sitenow_people_person_type_filter_select($element['field_person_types_target_id']);
+        // Show/hide person type statuses based on person type selections.
+        $element['field_person_types_target_id']['#ajax'] = [
+          'callback' => '_person_types_callback',
+          'event' => 'change',
+          'wrapper' => 'form-ajax-wrapper',
+          'speed' => 'fast',
+          'progress' => [
+            'type' => 'throbber',
+            'message' => t('Please wait...'),
+          ],
+        ];
+
+        $person_type_options = $element['field_person_types_target_id']['#options'];
+        $person_type_status_options = $element['field_person_type_status_value']['#options'];
+
+        foreach ($person_type_options as $key => $option) {
+          if (isset($person_type_status_options[$key])) {
+            $element['field_person_type_status_value']['#states']['visible'][] = [
+              ':input[name="settings[override][exposed_filters][field_person_types_target_id]"]' => [
+                'value' => $key,
+              ],
+            ];
+          }
+        }
+
+        $type = $form_state->getValue([
+          'settings',
+          'override',
+          'exposed_filters',
+          'field_person_types_target_id',
+        ]);
+
+        if (!$type && !empty($element['field_person_types_target_id']['#default_value'])) {
+          $type = $element['field_person_types_target_id']['#default_value'];
```

Comment:
It looks like when the field_person_types_target_id changes in an open form, the `$form_state->getValue(...)` ends up returning `null`, and so it updates the type to the default value, which is whatever the form was previously saved as. Still unsure why the field appears to update correctly, but then submit with this reverted value.

Reply:
I made an update to the `$type` setting and checking. It functions as expected now, but will throw a form value warning when a person type status is set and person type changes (say, "Former Graduate Assistant" is selected and person type is changed to "Faculty"--the person type status options will update to "Any" and "Former Faculty", and can be submitted correctly, but a warning that "An illegal choice has been detected. Please contact the site administrator." will be displayed on the field.

Reply:
The fix here feels wrong to me, as the comments above the `$form_state->getUserInput()` method seem to indicate that there is almost no reason to use it instead of `$form_state->getValues()` instead:
```
   * These are raw and unvalidated, so should not be used without a thorough
   * understanding of security implications. In almost all cases, code should
   * use self::getValues() and self::getValue() exclusively.
```

@cory-skeers Since we are calling the `getValue` version of this directly above that and it is failing to produce the correct value sometimes, that leads me to believe that there is additional processing necessary in `person_types_callback` to ensure that there is alignment between the values expected and the values provided:
https://github.com/uiowa/uiowa/blob/02ed5a20c64decd337254fa894c1b4afd2842e9c/docroot/modules/custom/sitenow_people/sitenow_people.module#L249-L251

I am in agreement that we should figure this out before releasing it because it indicates a potentially larger problem in the way that we are handling the AJAX callback that could be more problematic than what we have discovered so far.